### PR TITLE
Updates to "Working with Tables" - Adds info on "index renaming" and the table command

### PR DESCRIPTION
**Pull request by NotTheDr01ds**
Implements #1574 

Also, while there, I noticed that the chapter on Tables doesn't have any info on the `table` command, which seemed a bit odd - Added some detail regarding it, since it sometimes causes confusion.

### CODE REVIEW

File: book/working_with_tables.md [hunk 1]
```diff
@@ -390,3 +394,191 @@ You can also [`reject`](/commands/docs/reject.md) columns in a table by passing
...
+# => ╰────┴──────────┴──────┴─────────┴──────────────╯
+```
+
+The `sort-by modified` now _also_ sorts the `index` along with the rest of the columns.
+
+##### Adding a Row Header
+
+```nu
+let table = [
+[additions | deletions | delta ];
```
Comment (fdncred) [inline book/working_with_tables.md:515]: I don't think we should publicize `|` working as a delimiter here. I think it's a bug and probably shouldn't be allowed.

Comment (NotTheDr01ds) [reply book/working_with_tables.md:515]: You're right, of course.  I've gotten used to it so quickly for table literals that I've almost forgotten it's not supported.  It's definitely a bug, as the parse code specifically *doesn't* include `'|` as a list separator today.  But I'm going to propose that we just fix the bug by changing the parser to actually allow it.  I already have a fix in and tested, so I'll drop a PR soon and see if it flies.

In the meantime, I agree that we shouldn't document it until and unless it's official, so I'll reformat this.

**Comment by NotTheDr01ds** [comment]
@fdncred Thanks! Updated.

**Comment by fdncred** [comment]
Thanks

**Pull request merged**